apps/nextjs-website/react-components/components/PreFooter/PreFooter.tsx

```diff
@@ -0,0 +1,100 @@
+import { Box, Container, Stack, Typography } from '@mui/material';
+import Button from '@mui/material/Button';
+import Image from 'next/image';
+import { BackgroundColor } from '../common/Common.helpers';
+import { isJSX } from '../../types/common/Common.types';
+import { PreFooterProps } from '@react-components/types/PreFooter/PreFooter';
+import { useTheme, useMediaQuery } from '@mui/material';
+import googleBadgeLightBase64 from './BadgeImages/googleBadgeLightBase64';
+import appleBadgeLightBase64 from './BadgeImages/appleBadgeLightBase64';
+import googleBadgeDarkBase64 from './BadgeImages/googleBadgeDarkBase64';
+import appleBadgeDarkBase64 from './BadgeImages/appleBadgeDarkBase64';
+
+const styles = {
+  main: (isSmallScreen: boolean) => ({
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    flexDirection: isSmallScreen ? 'column' : 'row',
+    padding: { md: '64px 24px', xs: '32px 24px' },
+  }),
+  backgroundImage: (isSmallScreen: boolean, theme: 'light' | 'dark', decorationUrl: string) => ({
+    backgroundColor: theme === 'dark' ? '#031344' : BackgroundColor(theme),
+    backgroundImage: `url(${decorationUrl})`,
+    backgroundSize: isSmallScreen ? 'cover' : '30%',
+    backgroundPosition: isSmallScreen ? 'center' : 'right',
+    backgroundRepeat: 'no-repeat',
+  }),
+};
+
+const PreFooter = (props: PreFooterProps) => {
+  const { theme, title, decoration, storeButtons } = props;
+  const muiTheme = useTheme();
+  const isSmallScreen = useMediaQuery(muiTheme.breakpoints.down('sm'));
+  
+  const decorationUrl = decoration && 'url' in decoration ? decoration.url : '';
+  
+  const googleBadgeBase64 = theme === 'dark' ? googleBadgeLightBase64 : googleBadgeDarkBase64;
+  const appleBadgeBase64 = theme === 'dark' ? appleBadgeLightBase64 : appleBadgeDarkBase64;
+
+  return (
+    <Box component='section' sx={styles.backgroundImage(isSmallScreen, theme, decorationUrl)}>
+      <Container>
+        <Box sx={styles.main(isSmallScreen)}>
+          {decoration && isJSX(decoration) ? decoration : null}
```

Comment:
Has `decoration` been tested?

From what I understand this would return `null` on anything except for a JSX decoration, but its typing suggests it can also be an object containing a `url` field or an ImgProps object (which is not JSX).

These two cases would both return null, wouldn't they?